src/log.rs

```diff
@@ -1,3 +1,27 @@
+// Utility function to provide typed checking of the mask's field state.
```

Comment:
nit: in rust docs-string is `///`, `//` - a comment.  (if you render documentation `check_mask` would not have documentation.

Reply:
check_mask is private. Doctest can't access private methods from what I've read.

Reply:
Will private functions still get documentation rendered?

Reply:
anyways, if your intention is to document then probably is best to use `///` over '//'